install-yarn.js

```diff
@@ -6,13 +6,9 @@ var yarnBin
   , node = process.argv[0]
   ;
 
-if (process.env['npm_execpath'] && process.env['npm_execpath'].match(/node_modules[\/\\]yarn[\/\\]bin[\/\\]yarn\.js$/)) {
+if (process.env['npm_execpath'] && process.env['npm_execpath'].match(/[\/\\]yarn(-.+)?[\/\\]bin[\/\\]yarn\.js$/)) {
```

Comment:
What is the `(-.+)?` looking for here?

Reply:
As the path can be `/opt/yarn-v1.22.17/bin/yarn.js`, I add an optional group to append the `-vx.xx.xx`